plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -477,6 +477,43 @@ public Object[][] dbRefProvider()
         };
     }
 
+    @Test
+    public void testDbRefFieldOrder()
+    {
+        // DBRef's field order is databaseName, collectionName and id
+        // Create a table with different order and verify the result
+        String tableName = "test_dbref_field_order" + randomNameSuffix();
+        assertUpdate("CREATE TABLE test." + tableName + "(x row(id int, \"collectionName\" varchar, \"databaseName\" varchar))");
+
+        Document document = new Document()
+                .append("x", new DBRef("test_db", "test_collection", 1));
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        assertThat(query("SELECT * FROM test." + tableName))
+                .matches("SELECT CAST(row(1, 'test_collection', 'test_db') AS row(id int, \"collectionName\" varchar, \"databaseName\" varchar))");
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test
+    public void testDbRefMissingField()
```

Comment:
Can we also check if the fields are of different type ? And if we have three fields with different names ?

Reply:
Updated.